addons/binding/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/core/TelldusCoreBridgeHandler.java

```diff
@@ -78,8 +78,14 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     @Override
     public void dispose() {
         logger.debug("Telldus Core Handler disposed.");
-        deviceController.dispose();
-        eventHandler.remove();
+        if (deviceController != null) {
+            deviceController.dispose();
+            deviceController = null;
+        }
+        if (eventHandler != null) {
+            eventHandler.remove();
```

Comment:
@jarlebh do you mean the comment on/around this lines?

```
if (eventHandler != null) {
    eventHandler.removeListener((TelldusCoreDeviceController) deviceController);
    eventHandler.removeListener(this);
    eventHandler.remove();
    eventHandler = null;
}
```

Reply:
@jarlebh do you mean that 

`eventHandler.remove();`

should be replaced by:

```
eventHandler.removeListener((TelldusCoreDeviceController) deviceController);
eventHandler.removeListener(this);
eventHandler.remove();
```

Can you explain the difference?

Reply:
I just noticed that eventHandler.remove() does not actually remove the listeners it just removes the callbacks in the JNA/Telldus library. This would help the garbage collection, but not strictly needed since we recreate the eventHandler.

Reply:
@jarlebh can you do a follow up PR later?